wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -201,26 +201,29 @@ actual abstract class ProtoAdapter<E> actual constructor(
 
     // Obsolete; for Java classes generated before typeUrl and syntax were added.
     @JvmStatic fun <M : Message<M, B>, B : Message.Builder<M, B>> newMessageAdapter(
```

Comment:
As the comment pointed out, some methods are here because some old generated classes might use the more recent runtime. We need to keep the signatures as is.
We wanna keep all current `newMessageAdapter`s and add one with the new signature.

```kotlin
    @JvmStatic fun <M : Message<M, B>, B : Message.Builder<M, B>> newMessageAdapter(
      type: Class<M>,
      typeUrl: String,
      syntax: Syntax,
      loader: ClassLoader,
    )
```